### PR TITLE
program for fibonacci search

### PR DESCRIPTION
**Pull request by lovish318**
This Pull Request resolves Issue #19 

Description:
In this Pull Request we have done the following additions/edits for the code:

program for fibonacci search

### CODE REVIEW

**Comment by shyamdr** [comment]
#include<iostream>
 
using namespace std;
 
void FibonacciSearch(int *a, int start, int end, int *fab, int index, int item)
{
	int i, mid;
 
	// Assigning middle of the array using Fibonacci element.
	mid = start+fab[index-2];
 
	// Return if item found at mid index.
	if(item == a[mid])
	{
		cout<<"\n item found at "<<mid<<" index.";
		return;
	}
	// Return if item found at start index.
	else if(item == a[start])
	{
		cout<<"\n item found at "<<start<<" index.";
		return;
	}
	// Return if item found at end index.
	else if(item == a[end])
	{
		cout<<"\n item found at "<<end<<" index.";
		return;
	}
	// If mid becomes start or end of the sub-array then element not found.
	else if(mid == start || mid == end)
	{
		cout<<"\nElement not found";
		return;
	}
	// According to the item value choose the partion to proceed further.
	else if(item > a[mid])
		FibonacciSearch(a, mid, end, fab, index-1, item);
	else
		FibonacciSearch(a, start, mid, fab, index-2, item);
}
 
main()
{
	int n, i, biter, fab[20], a[20]={1, 9, 18, 24, 27, 35, 38, 41, 49, 53, 55, 66, 67, 72, 75, 77, 81, 89, 90, 97};
	char ch;
 
	fab[0] = 0;
	fab[1] = 1;
	i = 1;
	while(fab[i] < 20)
	{
		i++;
		fab[i] = fab[i-1]+fab[i-2];
	}
 
	up:
	cout<<"\nEnter the Element to be searched: ";
	cin>>n;
 
	// Implement Fibonacci search.
	FibonacciSearch(a, 0, 19, fab, i, n);
 
	// Ask user to enter choice for further searching.
	cout<<"\n\n\tDo you want to search more...enter choice(y/n)?";
	cin>>ch;
	if(ch == 'y' || ch == 'Y')
		goto up;
 
	return 0;
}

File: fibonacci_search/lovish318.c [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    return -1; // not found
+}
+
+int main()
+{
+    int data[]= {1, 4, 5, 7, 9, 11, 13, 16, 18, 20, 25, 27, 30, 32, 33, 36, 39, 41, 44, 47, 51, 53, 55};
+    int i, x, n;
+
+    x = 30;
```
Comment (utkarsh23) [inline fibonacci_search/lovish318.c]: Please make provisions to accept the array as well the element to be search for, as an input from the user.

**Comment by utkarsh23** [comment]
@shyamdr I'm not quite sure why you've commented whatever it is that you have commented. If you're looking to contribute to the repository, please create a pull request.

**Review by utkarsh23 — CHANGES_REQUESTED** [review]

**Comment by salman-bhai** [comment]
Are there any updates or should we close the PR?

**Comment by sbshah97** [comment]
PR closed due to inactivity

**Pull request closed without merging**